### PR TITLE
[CSL-1486] Update request parameters for search

### PR DESCRIPTION
**Pull request by stanlp1**
added pre filter, qs, offset.

Some weird lint fixes too.

### CODE REVIEW

**Review by jjl014 — APPROVED** [review]
Looking good to me! 🚀

File: spec/mocha.helpers.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ const getUserDefinedWindowProperties = () => {
   document.body.appendChild(iframe);
 
   const currentWindow = Object.getOwnPropertyNames(window);
-  const properties = currentWindow.filter((prop) => !Object.prototype.hasOwnProperty.call(iframe.contentWindow, prop));
+  const properties = currentWindow.filter(prop => !Object.prototype.hasOwnProperty.call(iframe.contentWindow, prop));
```
Comment (esezen) [inline spec/mocha.helpers.js:72]: Is there any chance your eslint version is out of sync? I'm getting an error with this new line and I think this line was specifically changed in the eslint update PR: https://github.com/Constructor-io/constructorio-client-javascript/pull/190

File: src/modules/search.js [hunk 2]
```diff
@@ -143,9 +159,11 @@ class Search {
    * @param {string} [parameters.sortOrder='descending'] - The sort order for results
    * @param {string} [parameters.section='Products'] - The section name for results
    * @param {object} [parameters.fmtOptions] - The format options used to refine result groups
```
Comment (esezen) [inline src/modules/search.js:161]: Can we add `Please refer to https://docs.constructor.io/rest_api/search/queries for details`?

File: spec/mocha.helpers.js (hunk 1 above)
Comment (stanlp1) [reply spec/mocha.helpers.js:72]: Hmm I think I dind't run npm install so I will redo these lint changes.

File: src/modules/search.js (hunk 2 above)
Comment (stanlp1) [reply src/modules/search.js]: Done 👍

File: src/modules/search.js [hunk 3]
```diff
@@ -142,10 +158,12 @@ class Search {
    * @param {string} [parameters.sortBy='relevance'] - The sort method for results
```
Comment (esezen) [inline src/modules/search.js:158]: Can we add `offset` here? (not here, after page but GH doesn't let me add a comment there)

**Review by esezen — APPROVED** [review]
LGTM!

**Pull request merged**